Source/EvolveM.cpp

```diff
@@ -21,164 +21,143 @@ void Compute_LLG_RHS(
-                        Hy_eff += Hy_exchange(i,j,k);
-                        Hz_eff += Hz_exchange(i,j,k);
-                    }
-                 
-                    if(DMI_coupling == 1)
-                    { 
-                        Hx_eff += Hx_DMI(i,j,k);
-                        Hy_eff += Hy_DMI(i,j,k);
-                        Hz_eff += Hz_DMI(i,j,k);
-                    }
-                 
-                    if(anisotropy_coupling == 1)
-                    {
-                        Hx_eff += Hx_anisotropy(i,j,k);
-                        Hy_eff += Hy_anisotropy(i,j,k);
-                        Hz_eff += Hz_anisotropy(i,j,k);
-                    }
-
-                   //Update M
-
-                   // 0 = unsaturated; compute |M| locally.  1 = saturated; use M_s
-                   amrex::Real M_magnitude = (M_normalization == 0) ? std::sqrt(std::pow(Mx_old(i, j, k), 2._rt) + std::pow(Mx_old(i, j, k), 2._rt) + std::pow(Mx_old(i, j, k), 2._rt))
-                                                              : Ms_arr(i,j,k); 
-
-                   // x component on x-faces of grid
-                   LLG_rhs_x(i, j, k) = LLG_RHS_x(Mx_old, My_old, Mz_old, alpha_arr, gamma_arr, M_magnitude, mu0, Hx_eff, Hy_eff, Hz_eff, i, j, k);
-
-                   // y component on x-faces of grid
-                   LLG_rhs_y(i, j, k) = LLG_RHS_y(Mx_old, My_old, Mz_old, alpha_arr, gamma_arr, M_magnitude, mu0, Hx_eff, Hy_eff, Hz_eff, i, j, k);
-
-                   // z component on x-faces of grid
-                   LLG_rhs_z(i, j, k) = LLG_RHS_z(Mx_old, My_old, Mz_old, alpha_arr, gamma_arr, M_magnitude, mu0, Hx_eff, Hy_eff, Hz_eff, i, j, k);
-
-                 }   
- 
-            });     
-        }
+                    Hx_eff += Hx_demag(i,j,k);
+                    Hy_eff += Hy_demag(i,j,k);
+                    Hz_eff += Hz_demag(i,j,k);
+                }
+                if(exchange_coupling == 1)
+                { 
+                    Hx_eff += Hx_exchange(i,j,k);
+                    Hy_eff += Hy_exchange(i,j,k);
+                    Hz_eff += Hz_exchange(i,j,k);
+                }
+             
+                if(DMI_coupling == 1)
+                { 
+                    Hx_eff += Hx_DMI(i,j,k);
+                    Hy_eff += Hy_DMI(i,j,k);
+                    Hz_eff += Hz_DMI(i,j,k);
+                }
+             
+                if(anisotropy_coupling == 1)
+                {
+                    Hx_eff += Hx_anisotropy(i,j,k);
+                    Hy_eff += Hy_anisotropy(i,j,k);
+                    Hz_eff += Hz_anisotropy(i,j,k);
+                }
+                //Update M
+                // 0 = unsaturated; compute |M| locally.  1 = saturated; use M_s
+                amrex::Real M_magnitude = (M_normalization == 0) ? std::sqrt(std::pow(Mx_old(i, j, k), 2._rt) + std::pow(My_old(i, j, k), 2._rt) + std::pow(Mz_old(i, j, k), 2._rt))
+                                                           : Ms_arr(i,j,k); 
```

Comment:
bug in computing M magnitude

Reply:
Thanks for catching that! These type of typos are hardest to find